exercises/binary-search/example.scala

```diff
@@ -0,0 +1,23 @@
+import scala.annotation.tailrec
+
+object BinarySearch {
+  def search[T](seq: Seq[T], value: T)(implicit ord: T => Ordered[T]): Option[Int]
```

Comment:
I originally wrote the tests to use Seq. But, all the test case comments from the canonical-data.json use the term Array. It felt wrong to have the test code use Seq, with comments using Array.

I suppose we could use Seq and add something to the readme. Or, the test generator could modify the test case comments to use "seq" instead of "array". Or, we can just leave the test impl to use Array.

On the template or .keep. I added an empty test file BinarySearch.scala instead of .keep.

On a separate note - I am going to be unreachable until Thursday morning US time. 

Thanks for all your input!